src/util/StringStream.js

```diff
@@ -73,9 +74,14 @@ class StringStream {
   current(){return this.string.slice(this.start, this.pos)}
   hideFirstChars(n, inner) {
```

Comment:
I can't see any difference in the pug mode when changing sol() to "sol() {return this.pos == 0}" and looking through the code I can't see anything which should be affected, but I can't say I'm very familiar with the pug syntax or how which test cases would a be a good fit. 

Another solution is adding this to the soy mode, where it eats indentation before entering the inner mode to set it to the correct stream position:

      if (stream.sol()) {
        for (var indentEaten = 0; indentEaten < state.indent; indentEaten++) {
          if (!stream.eat(/\s/)) break;
        }
        if (indentEaten > 0) return null;
      }